src/jshint.js

```diff
@@ -4247,12 +4305,12 @@ var JSHINT = (function() {
     x.lbp = 25;
   }(prefix("yield", function() {
     var prev = state.tokens.prev;
-    if (state.inESNext(true) && !state.funct["(generator)"]) {
+    if (state.inES6(true) && !state.funct["(generator)"]) {
```

Comment:
Okay,  im doubting myself now,  but anyway the gyst is... 
Any es7 features should only be on in esversion = 7 and esnext should map to esversion 7. That way esversion is "correct" and it is not a breaking change..